src/main/java/com/eprosima/fastdds/idl/templates/DDSPubSubTypeHeader.stg

```diff
@@ -74,6 +74,36 @@ $typedefs : { typedef |typedef $typedef.typedefContentTypeCode.cppTypename$ $typ
 >>
 
 struct_type(ctx, parent, struct) ::= <<
+
+$if(struct.isPlain)$
+namespace detail {
+
+    template<typename Tag, typename Tag::type M>
+    struct $struct.name$_rob
+    {
+        friend constexpr typename Tag::type get(
+                Tag)
+        {
+            return M;
+        }
+    };
+
+    struct $struct.name$_f
+    {
+        typedef $last(struct.members).typecode.cppTypename$ $struct.name$::* type;
```

Comment:
What would happen if `struct.members` is empty?

Reply:
Fixed in 70a11b3